### ISSUE TITLE
New Lint/AmbiguousOperatorPrecedence: problems with shift operator

### ISSUE THREAD

**Issue by andrew-aladev** [rubocop/rubocop#10079]
Please create file `sample.rb`:

```ruby
a = 1 + 2 * (3 << 4)
```

```
rubocop --only="Lint/AmbiguousOperatorPrecedence" sample.rb
```

--------

## Expected behavior

Everything is fine.

## Actual behavior

```
W: [Correctable] Lint/AmbiguousOperatorPrecedence: Wrap expressions with varying precedence with parentheses to avoid ambiguity.
a = 1 + 2 * (3 << 4)
        ^^^^^^^^^^^^
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.21.0 (using Parser 3.0.2.0, rubocop-ast 1.11.0, running on ruby 3.0.2 x86_64-linux)
```

See also #10047.

**Comment by dvandersluis** [comment]
I'm not sure what the bug is here? The offense is not because of the `<<` operator, but because of the `+` operator. Since `*` has higher precedence than `+`, the parentheses deambiguate the right side of the `+` operation.

The idea is that the parentheses will make it obvious that this expression can be expressed as:

```ruby
b = 2 * (3 << 4)
a = 1 + b
```

**Comment by andrew-aladev** [comment]
Ok, thank you. I am going to disable this cop.

So this is not a bug.
